src/components/popup.js

```diff
@@ -3,6 +3,7 @@ class Popup {
     this.popupSelector = popupSelector;
     this.popupElement = document.querySelector(this.popupSelector);
     this._handleEscClose = this._handleEscClose.bind(this);
+    this.setEventListeners();
```

Comment:
` this.setEventListeners();` is not needed inside `Popup` because it has `children` that reassign this method (calling `super.setEventListeners()` inside it). So, you don't need to duplicate calling it here and there

Reply:
done

Reply:
> SetEventlisteners inside the Popup class is responsible for closing any popup with the use of Esc button or a click to any zone outside of the popup. So once popup class is called - any popup could be closed that way, using this method. This is why it's needed there

You have `this._setEventListeners();` inside the `children`, where you have `super.setEventListeners();` which adds the same closing handlers for `Escape` and `overlay`, so you'll get 2 same listeners for each popup